File: BE-Task_1-Java/src/main/java/io/mekari/assessment/refundbackendjava/repository/RefundRepository.java
package io.mekari.assessment.refundbackendjava.repository;

import io.mekari.assessment.refundbackendjava.domain.model.Refund;
import java.util.List;
import java.util.Optional;

public interface RefundRepository {
    void create(Refund refund);

    Optional<Refund> findById(String refundId);

    List<Refund> listAll();
}

File: BE-Task_1-Ruby/internal/repository/memory/refund_repository.rb
module RefundBackend
  module Repository
    module Memory
      class RefundRepository < RefundBackend::Repository::Interfaces::RefundRepository
        def initialize(store)
          @store = store
        end

        def create(refund)
          @store.insert_refund(refund)
        end

        def find_by_id(refund_id)
          @store.fetch_refund(refund_id)
        end

        def list_all
          @store.list_refunds
        end
      end
    end
  end
end
